packages/system/src/Item/api/api.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Item,
+  SortBy,
+  ItemSortFieldInput,
+  FilterBy,
+} from '@openmsupply-client/common';
+import {
+  ItemRowFragment,
+  ItemsListViewQuery,
+  ItemFragment,
+} from './operations.generated';
+import { ItemApi } from './hooks';
+import { getItemSortField } from '../utils';
+
+const itemsGuard = (itemsQuery: ItemsListViewQuery) => {
+  if (itemsQuery.items.__typename === 'ItemConnector') {
+    return itemsQuery.items;
+  }
+
+  throw new Error(itemsQuery.items.error.description);
+};
+
+export const ItemQueries = {
```

Comment:
looking at this now, maybe it should be

```
const getItemQueries = (api, storeID) => ({ ... })..
```

And then 

```
const useItemApi = () => {
    const storeId = getStoreId()
    const  {client} = getClient()
    const sdk = getSdk(client)
    return getItemQueries(storeId, sdk)
}
```


and is probably a pattern to use elsewhere

Reply:
yes, that's the pattern I'm ending up with - I think it's about the best so far